src/main/java/org/truffleruby/interop/InteropNodes.java

```diff
@@ -689,24 +690,31 @@ public boolean hasKeys(Object receiver,
 
     }
 
-    @CoreMethod(names = "keys", isModuleFunction = true, required = 1)
+    @CoreMethod(names = "keys", isModuleFunction = true, required = 1, optional = 1)
     public abstract static class KeysNode extends CoreMethodArrayArgumentsNode {
 
+        public abstract Object executeSize(VirtualFrame frame, TruffleObject receiver, Object internal);
```

Comment:
It should be named `executeKeys` and the methods below `keys`.

Reply:
Done.